pkg/agent/cniserver/pod_configuration.go

```diff
@@ -677,12 +677,21 @@ func (pc *podConfigurator) recordPodEvent(ifConfig *interfacestore.InterfaceConf
 }
 
 func (pc *podConfigurator) processPortStatusMessage(status *openflow15.PortStatus) {
-	// Update Pod OpenFlow entries only after the OpenFlow port state is live.
-	if status.Desc.State != openflow15.PS_LIVE {
+	ofPort := status.Desc.PortNo
+	state := status.Desc.State
+	// Update Pod OpenFlow entries only after the OpenFlow port state is live or down.
+	// Accepting Port state "openflow15.PS_LINK_DOWN" is a workaround for Windows OVS issue https://github.com/openvswitch/ovs-issues/issues/351.
+	// In which OVS does not correctly implement function netdev_windows_update_flags, so OVS doesn't update ifp_flags
+	// after a new OpenFlow port is successfully installed. Since this OVS issue doesn't have side impact on datapath
+	// packets forwarding, antrea-agent will ignore the bad state to ensure the Pod's OpenFlow entries are installed as
+	// long as the port number is allocated.
+	if status.Desc.State != openflow15.PS_LIVE && status.Desc.State != openflow15.PS_LINK_DOWN {
```

Comment:
When `openflow15.PS_LINK_DOWN` is received, is it guaranteed that ofPort is valid?

Reply:
OVS has guaranteed the port number is valid. By analyzing the messages on live testbed, OVS may only send a single PortStatus message when it uses state "PS_LINK_DOWN", unlike the desired state in which OVS should send two messages, one is using state value 0, and the other uses PS_LIVE; in this case, OVS only sends one message, and the PortNo in the message is a valid value which could be used as an identifier in the OpenFlow switch.

In fact this OVS issue exists for long time. Previously we only read the OF port value from OVSDB, and doesn't verify the port state as what we do now.

Reply:
Added an additional check on the port number, which may ignore the message if the number is 0.